src/devices/LiquidLevelSwitch/LiquidLevelSwitchBase.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+
+namespace Iot.Device.LiquidLevelSwitch
+{
+    /// <summary>
+    /// Represents base class for Liquid Level Switches
+    /// </summary>
+    public abstract class LiquidLevelSwitchBase : IDisposable
+    {
+        private readonly int _pin;
+        private readonly PinMode _switchMode;
+        private readonly bool _shouldDispose;
+
+        private GpioController _controller;
+
+        /// <summary>Initializes a new instance of the <see cref="LiquidLevelSwitchBase" /> class.</summary>
+        /// <param name="pin">The pin.</param>
+        /// <param name="gpioController">The gpio controller.</param>
+        /// <param name="switchMode">The switch mode.</param>
```

Comment:
this is not obvious how it works, perhaps something like `liquidPresentPinState`